src/doc/src/commands/cargo-publish.md

```diff
@@ -24,8 +24,8 @@ following steps:
    and may timeout. In that case, you will need to check for completion
    manually. This timeout does not affect the upload.
 
-This command requires you to be authenticated with either the `--token` option
-or using [cargo-login(1)](cargo-login.html).
+This command requires you to be authenticated using [cargo-login(1)](cargo-login.html) 
+or [environment variables](../reference/environment-variables.html).
```

Comment:
Probably instead of pointing to the entire env doc, we make the link a bit precise to relevant config fields. What do you think?

```suggestion
or environment variables of the [`registry.token`](../reference/config.html#registrytoken)
and [`registries.<name>.token`](../reference/config.html#registriesnametoken) config fields.
```

Though I am still wanting something similar to this happening: <https://rust-lang.zulipchat.com/#narrow/channel/246057-t-cargo/topic/Anchors.20for.20each.20environment.20variable.20in.20The.20Cargo.20Book>

Reply:
Thanks for the edit, I missed the fact that config page has direct mentions of env variables with links 😥 

As for 
> This in both registry.token and registries.<name>.token doc might need to remove

I would not do it in this PR, because `--token` is still used in `yank` and `owner` commands. Let's postpone it until the deprecation is done in all places.

Reply:
That is fair. Thank you!

Reply:
Created an issue tracking it: <https://github.com/rust-lang/cargo/issues/16049>